src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs

```diff
@@ -23,80 +23,96 @@ namespace NLog.Web.LayoutRenderers
     /// </summary>
 #else
     /// <summary>
-    /// Rendering Application BasePath. <see cref="HostingEnvironment.MapPath"/>("~")
+    /// Rendering Application BasePath. <see cref="IHostEnvironment.MapPath"/>("~")
     /// </summary>
 #endif
     [LayoutRenderer("aspnet-appbasepath")]
     [ThreadAgnostic]
     public class AspNetAppBasePathLayoutRenderer : LayoutRenderer
     {
 #if ASP_NET_CORE
-        private IHostEnvironment HostEnvironment => _hostEnvironment ?? (_hostEnvironment = ServiceLocator.ResolveService<IHostEnvironment>(ResolveService<IServiceProvider>(), LoggingConfiguration));
         private IHostEnvironment _hostEnvironment;
 
-        private string AppBasePath => _appBasePath ?? ResolveAppBasePath(HostEnvironment?.ContentRootPath, out _appBasePath);
-        private string _appBasePath;
+        /// <summary>
+        /// Provides access to the current IHostEnvironment
+        /// </summary>
+        /// <returns>IHostEnvironment or <c>null</c></returns>
+        internal IHostEnvironment HostEnvironment
+        {
+            get => _hostEnvironment ?? (_hostEnvironment = RetrieveHostEnvironment(ResolveService<IServiceProvider>(), LoggingConfiguration));
+            set => _hostEnvironment = value;
+        }
+
+        internal static IHostEnvironment RetrieveHostEnvironment(IServiceProvider serviceProvider, LoggingConfiguration loggingConfiguration)
+        {
+            return ServiceLocator.ResolveService<IHostEnvironment>(serviceProvider, loggingConfiguration);
+        }
 #else
-        private string AppBasePath => _appBasePath ?? ResolveAppBasePath(HostingEnvironment.MapPath("~"), out _appBasePath);
-        private static string _appBasePath;
+        /// <summary>
+        /// Provides access to the current IHostEnvironment
+        /// </summary>
+        /// <returns>IHostEnvironment or <c>null</c></returns>
+        internal IHostEnvironment HostEnvironment { get; set; } = Internal.HostEnvironment.Default;
```

Comment:
Possible to merge the HostEnvironment-properties together, so the only difference is the method used to resolve the `IHostEnvironment`-implementation?